src/Runner.Common/JobServerQueue.cs

```diff
@@ -437,18 +455,21 @@ private async Task ProcessFilesUploadQueueAsync(bool runOnce = false)
             }
         }
 
-        private async Task ProcessSummaryUploadQueueAsync(bool runOnce = false)
+        private async Task ProcessResultsUploadQueueAsync(bool runOnce = false)
         {
             Trace.Info("Starting results-based upload queue...");
 
             while (!_jobCompletionSource.Task.IsCompleted || runOnce)
             {
-                List<SummaryUploadFileInfo> filesToUpload = new();
-                SummaryUploadFileInfo dequeueFile;
-                while (_summaryFileUploadQueue.TryDequeue(out dequeueFile))
+                List<ResultsUploadFileInfo> filesToUpload = new();
+                ResultsUploadFileInfo dequeueFile;
+                while (_resultsFileUploadQueue.TryDequeue(out dequeueFile))
                 {
-                    filesToUpload.Add(dequeueFile);
-                    // process at most 10 file upload.
+                    if (_resultsClientInitiated)
```

Comment:
do you still need this if we skip it in the enqueue side already?

Reply:
hmm, not really, I can remove this.  I just feel it's safer to check the condition right before we act on it, but since the queue is a private member variable, I guess it's okay to just guard it at enqueue time.